### ISSUE TITLE
DynamoDB: Add support for delete protection flag

### ISSUE THREAD

**Issue by allanchua101** [aws/aws-cdk#24568]
### Describe the issue

Please add support for delete protection flag. AWS CLI and CloudFormation documentations now reflect the property's support. Hoping to see it working on CDK too!

### Links

[AWS Announcement](https://aws.amazon.com/blogs/database/how-to-use-deletion-protection-to-enhance-your-amazon-dynamodb-table-protection-strategy/)
[DynamoDB Typescript CDK Doc](
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-dynamodb.TableOptions.html)

**Comment by pahud** [comment]
Yes, please stay tuned on https://github.com/aws/aws-cdk/issues/24540
